### ISSUE TITLE
Java code for enum list input produces invalid output

### ISSUE THREAD

**Issue by hannesj** [dotansimha/graphql-code-generator#7002]
**Describe the bug**

The `java` type generator produces code, which crashes during runtime due to being unable to cast a `String` into the generated `Enum` type, if it is given to a data fetcher as an input list.

**To Reproduce**

1. My GraphQL schema:

```graphql
schema {
  query: Query
}

type Query {
  authorize(roles: [Role]): Boolean
}

enum Role {
  USER,
  ADMIN
}
```

. My `codegen.yml` config file:

```yml
generates:
  LegacyGraphQLTypes.java:
    plugins:
      - java
```

**Current behavior**

The codegen produces the following code, which casts a `Iterable<String>` into a `Iterable<Role>`, which compiles nicely, but throws an exception, when you try to use the value.

```java
package .;

import java.util.HashMap;
import java.util.Map;
import java.util.List;
import java.util.stream.Collectors;

public class Types {
  public static class QueryAuthorizeArgs {
    private Iterable<Role> roles;
  
    public QueryAuthorizeArgs(Map<String, Object> args) {
      if (args != null) {
        if (args.get("roles") != null) {
        		this.roles = (Iterable<Role>) args.get("roles");
        }
      }
    }
  
    public Iterable<Role> getRoles() { return this.roles; }
    public void setRoles(Iterable<Role> roles) { this.roles = roles; }
  }
  public enum Role {
    USER,
    ADMIN
    
  }
}
```

If we change the input parameter to be an instance of `Role`, it produces the correct output, where it tries to map the enum from the string representation.

```java
package .;

import java.util.HashMap;
import java.util.Map;

public class Types {
  public static class QueryAuthorizeArgs {
    private Role roles;
  
    public QueryAuthorizeArgs(Map<String, Object> args) {
      if (args != null) {
        if (args.get("roles") instanceof Role) {
          this.roles = (Role) args.get("roles");
        } else {
          this.roles = Role.valueOf((String) args.get("roles"));
        }
      }
    }
  
    public Role getRoles() { return this.roles; }
    public void setRoles(Role roles) { this.roles = roles; }
  }
  public enum Role {
    USER,
    ADMIN
    
  }
  
}

```

**Environment:**

Playground at https://www.graphql-code-generator.com/

**Comment by Urigo** [comment]
Hi @hannesj and thank you for the report

Sorry but I'm not adding a lot here but just labeling it according to our new Contribution Guide and [issue flow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

It seems already got into `stage 1` thanks to your reproduction! Thank you for that!

Now in order to advance to `stage 2` we'll need a failing test, would be great if someone could help progress the issues through the [stages](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow).

Thank you and sorry that this comment is not a complete solution (yet).

**Comment by leonardehrenfried** [comment]
Is https://github.com/dotansimha/graphql-code-generator/blob/master/packages/plugins/java/java/tests/java.spec.ts the correct place to put the test?

**Comment by Urigo** [comment]
yes!

**Comment by optionsome** [comment]
I created a pull request that contains both a test and a fix for this problem. Waiting for it to be reviewed by someone

**Comment by charlypoly** [comment]
Thanks to @optionsome contribution (👏🏼 ), this issue is now fixed in `@graphql-codegen/java@3.2.4`
